core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorer.kt

```diff
@@ -273,6 +273,8 @@ private class InfraExplorerImpl(
         var pathStarted = incrementalPath.pathStarted
 
         for (block in routeBlocks) {
+            blockRoutes.getOrPut(block) { mutableSetOf() }.add(route)
```

Comment:
Isn't there a risk of ending up in an inconsistent state (also related with the doc of the function)?

Reply:
Good catch, in fact there's some general issues in how we handle this map. I'll fix it